Assets/Scripts/Localization/LocalizationTable.cs

```diff
@@ -6,193 +6,118 @@
+            return GetLocalization(key, FallbackMode.ReturnDefaultLanguage, currentLanguage, additionalValues);
         }
 
-        /**
-         * <summary>
-         * Returns the localization for the given key, or the key itself, if no translation exists.
-         * </summary>
-         */
-        public static string GetLocalization(string key, FallbackMode fallbackMode, string language, params string[] additionalValues)
+        /// <summary>
+        /// Returns the localization for the given key, or the key itself, if no translation exists.
+        /// </summary>
+        private static string GetLocalization(string key, FallbackMode fallbackMode, string language, params string[] additionalValues)
         {
-            key = language + "_" + key;
             string value;
-            if (localizationTable.TryGetValue(key, out value))
+            if (localizationTable.ContainsKey(language) && localizationTable[language].TryGetValue(key, out value))
             {
                 return string.Format(value, additionalValues);
             }
-            else
+            if (!missingKeysLogged.Contains(key))
             {
-                if (!missingKeysLogged.Contains(key))
-                {
-                    missingKeysLogged.Add(key);
-                    Logger.LogWarning("Translation for " + key + " failed: Key not in dictionary.");
-                }
-
-                switch (fallbackMode)
-                {
-                    case FallbackMode.ReturnKey:
-                        if (additionalValues.Length >= 1)
-                        {
-                            return key + " " + additionalValues[0];
-                        }
+                missingKeysLogged.Add(key);
+                Logger.LogWarning(string.Format("Translation for {0} in {1} failed: Key not in dictionary.", key, language));
+            }
 
-                        return key;
-                    case FallbackMode.ReturnEnglish:
-                        return GetLocalization(key, FallbackMode.ReturnKey, "en_US", additionalValues);
-                    case FallbackMode.ReturnEmpty:
-                        return "";
-                    default:
-                        return "";
-                }
+            switch (fallbackMode)
+            {
+            case FallbackMode.ReturnKey:
+                return additionalValues != null && additionalValues.Length >= 1 ? key + " " + additionalValues[0] : key;
+            case FallbackMode.ReturnDefaultLanguage:
+                return GetLocalization(key, FallbackMode.ReturnKey, defaultLanguage, additionalValues);
+            default:
+                return string.Empty;
             }
         }
 
         public static string[] GetLanguages()
         {
-            return registeredLanguages.ToArray();
+            return localizationTable.Keys.ToArray();
         }
     }
-}
+}
```

Comment:
Ah, but thats where youre wrong, a line [is defined](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206) as:

> 3.206 Line
> A sequence of zero or more non- _newline_ characters plus a terminating _newline_ character.

From [Stackoverflow](http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline). In practice, this means that what you think is the last line isnt. Because it doesnt end with an end-of-line char. This obviously doesnt matter in most cases, or you would know about it. If you look at the Github diff of the file theres a small icon there highlighting the missing eol, and its not just for show.